### PR TITLE
chore(vpc): add scan unused services logic to VPC checks

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Add logic of unused services to VPC checks:

```
vpc_subnet_no_public_ip_by_default

vpc_subnet_separate_private_public

vpc_subnet_different_az
```

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/vpc/vpc_subnet_no_public_ip_by_default/vpc_subnet_no_public_ip_by_default.py [hunk 1]
```diff
@@ -7,17 +7,20 @@ def execute(self):
...
-                report.resource_id = subnet.id
-                report.resource_arn = subnet.arn
-                if subnet.mapPublicIpOnLaunch:
-                    report.status = "FAIL"
-                    report.status_extended = f"VPC subnet {subnet.name if subnet.name else subnet.id} assigns public IP by default."
-                else:
-                    report.status = "PASS"
-                    report.status_extended = f"VPC subnet {subnet.name if subnet.name else subnet.id} does NOT assign public IP by default."
-                findings.append(report)
+                print(subnet.in_use)
```
Comment (jfagoagas) [inline prowler/providers/aws/services/vpc/vpc_subnet_no_public_ip_by_default/vpc_subnet_no_public_ip_by_default.py]:
```suggestion
```

File: prowler/providers/aws/services/vpc/vpc_subnet_no_public_ip_by_default/vpc_subnet_no_public_ip_by_default.py [hunk 2]
```diff
@@ -7,17 +7,20 @@ def execute(self):
...
+                # Check if ignoring flag is set and if the VPC Subnet is in use
+                if vpc_client.provider.scan_unused_services or subnet.in_use:
```
Comment (jfagoagas) [inline prowler/providers/aws/services/vpc/vpc_subnet_no_public_ip_by_default/vpc_subnet_no_public_ip_by_default.py:11]: We need to add that to the documentation right?

Comment (MrCloudSec) [reply prowler/providers/aws/services/vpc/vpc_subnet_no_public_ip_by_default/vpc_subnet_no_public_ip_by_default.py:11]: Yes, done!

**Review by jfagoagas — DISMISSED** [review]
👍

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**